### ISSUE TITLE
"no rate limit information found" on Pleroma/Pixelfed, etc.

### ISSUE THREAD

**Issue by giocomai** [schochastics/rtoot#95]
I understand this is described as a package for the Mastodon API, but it appears this may work just as well with all other parts of the Fediverse relying on ActivityPub.

Indeed, the version 0.1.0 on CRAN seems to work nicely also with [Pleroma](https://pleroma.social/)  ~~and [Pixelfed](https://pixelfed.org/)~~, including calls that require authorization. 

The current development version (as of 2022-11-20) however doesn't, giving the error `"no rate limit information found"` for most calls. 

See e.g. 

```
get_timeline_public(instance = "pythondevs.social") # random Pleroma instance
get_timeline_public(instance = "pixelfed.social") # Main Pixelfed instance
```

I have checked the headers given by the Pleroma API, and indeed, no rate limit information is included, even if obviously they enforce a rate limit (perhaps it's given only when the rate limit is hit?).

I may try to troubleshoot this further to find out if this is exposed in some other format, but either way, I wonder if an error is really warranted when `no rate limit information found`. If no rate limit information is found, after all, I would expect that all is fine. 

For reference, this is [my current workaround](https://github.com/giocomai/rtoot/commit/4e16e2bc71e7dc33f289244a716e03607308dff1), which fixes this issue for me. I can of course turn this into a pull request, but I am not convinced this is the most polished way to go about this.

Perhaps, an additional parameter making it possible to toggle this would also be an alternative. 

I understand that this package targets Mastodon, so of course no complaints if you prefer to just close this issue. But since everything seems otherwise to work great across the non-Mastodon part of the Fediverse with this package, perhaps this can still be considered.

**Comment by schochastics** [comment]
Haven't looked at other fediverse instances yet. Happy that it seems to work so well already. I will think about including this in a later release. For now, we keep focusing on Mastodon

**Comment by schochastics** [comment]
@giocomai I removed the error now and just a warning is displayed.

`get_timeline_public(instance = "pythondevs.social")` works but 
`get_timeline_public(instance = "pixelfed.social")` throws a 500 error. I can;t find any information if it is even possible to get data from pixelfed. Let me know if you have any pointers

**Comment by chainsawriot** [comment]
@schochastics All pixelfed endpoints require authentication.

https://github.com/dansup/pixelfed-php

**Comment by schochastics** [comment]
ok so one does need an account there

**Comment by giocomai** [comment]
I confirm that this seems to work nicely with Pleroma.  Thanks a lot for taking care of this from a grateful Pleroma user. 

When I opened this issue I did quite a lot of troubleshooting with Pleroma, but only very quickly with Pixelfed... I now realise that probably nothing ever really worked with Pixelfed, beyond things such as `get_instance_general()`.

With Pixelfed there are authorization issues which seem to be related to the fact that Pixelfed does not currently support `urn:ietf:wg:oauth:2.0:oob`  as redirect uri, which it should, and is [a known issue](https://github.com/pixelfed/pixelfed/issues/2522). So that's something completely different.

If you don't mind, I 'll proceed and close this issue, as the specific matter it raised has now been solved. Thanks a lot!
